### PR TITLE
telegraf: add correct version info

### PR DESCRIPTION
**Pull request by Hipska**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Discussion: https://github.com/Homebrew/discussions/discussions/3710

This PR is aimed to let users know the correct telegraf version by running `telegraf --version`.

### CODE REVIEW

File: Formula/telegraf.rb [hunk 1]
```diff
@@ -23,7 +24,7 @@ class Telegraf < Formula
   depends_on "go" => :build
 
   def install
-    system "go", "build", *std_go_args, "-ldflags", "-X main.version=#{version}", "./cmd/telegraf"
+    system "go", "build", *std_go_args, "-ldflags", "-X github.com/influxdata/telegraf/internal.Version=#{version}-brew", "./cmd/telegraf"
```
Comment (bayandin) [inline Formula/telegraf.rb:27]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/influxdata/telegraf/internal.Version=#{version}-brew"), "./cmd/telegraf"
```

Comment (Hipska) [reply Formula/telegraf.rb:27]: Why the `-s -w`? They don't seem to be added at upstream as well?

Comment (SMillerDev) [reply Formula/telegraf.rb:27]: That's why we generally add them

File: Formula/telegraf.rb [hunk 2]
```diff
@@ -41,6 +42,7 @@ def post_install
   end
 
   test do
+    assert_match version.to_s, shell_output("#{bin/telegraf} --version")
```
Comment (gromgit) [inline Formula/telegraf.rb:45]:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/telegraf --version")
```

**Review by gromgit — CHANGES_REQUESTED** [review]

File: Formula/telegraf.rb (hunk 1 above)
Comment (Hipska) [reply Formula/telegraf.rb:27]: Will profiling then still work?

Comment (chenrui333) [reply Formula/telegraf.rb:27]: > `-s -w`

it is just for omitting the symbol table and debugging information, has nothing to do with the app.

Could not find the blog post.

Comment (chenrui333) [reply Formula/telegraf.rb:27]: at this phase, we should make it into the std_go_args and apply by default.

Comment (Hipska) [reply Formula/telegraf.rb:27]: > it is just for omitting the symbol table and debugging information, has nothing to do with the app.

It is exactly for this reason the telegraf team didn’t add these flags to their build. Right @jpowers @sspaink?

Comment (chenrui333) [reply Formula/telegraf.rb:27]: I think you confused the two things, the debugging information for the binary, not for the application it is trying to profiling against?

Comment (powersj) [reply Formula/telegraf.rb:27]: [We also include these](https://github.com/influxdata/telegraf/blob/master/Makefile#L479) when we build release artifacts.

File: Formula/telegraf.rb [hunk 3]
```diff
@@ -23,7 +24,7 @@ class Telegraf < Formula
   depends_on "go" => :build
 
   def install
-    system "go", "build", *std_go_args, "-ldflags", "-X main.version=#{version}", "./cmd/telegraf"
+    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/influxdata/telegraf/internal.Version=#{version}-brew"), "./cmd/telegraf"
```
Comment (SMillerDev) [inline Formula/telegraf.rb:27]: Why suffix the version?

File: Formula/telegraf.rb [hunk 4]
```diff
@@ -1,9 +1,10 @@
 class Telegraf < Formula
-  desc "Server-level metric gathering agent for InfluxDB"
-  homepage "https://www.influxdata.com/"
+  desc "Plugin-driven server agent for collecting & reporting metrics"
+  homepage "https://www.influxdata.com/time-series-platform/telegraf/"
   url "https://github.com/influxdata/telegraf/archive/v1.24.2.tar.gz"
   sha256 "e0b4bc65dc46a87b14a164c1226f6cee15fabfff14f355344bede3ef3e585925"
   license "MIT"
+  revision 1
```
Comment (SMillerDev) [inline Formula/telegraf.rb:7]: I don't think we need to force a reinstall for all users.
```suggestion
```

File: Formula/telegraf.rb (hunk 3 above)
Comment (Hipska) [reply Formula/telegraf.rb:27]: To be clear it is not the built version from Influxdata? I was going to add the revision number in this version as well. But your other review comment suggests not to add a revision?

File: Formula/telegraf.rb (hunk 4 above)
Comment (Hipska) [reply Formula/telegraf.rb:7]: Based on: https://docs.brew.sh/Formula-Cookbook#formulae-revisions

File: Formula/telegraf.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/telegraf.rb:27]: If you install it from Homebrew it'll never be the influxdata version. Do Debian etc suffix their Telegraf builds like this?

File: Formula/telegraf.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/telegraf.rb:7]: > to either ensure formulae continue to function as expected or to close a security issue.

Neither is really applicable here

Comment (Hipska) [reply Formula/telegraf.rb:7]: What about

> In Homebrew we sometimes accept formulae updates that don’t include a version bump. These include resource updates, new patches or fixing a security issue with a formula.

File: Formula/telegraf.rb (hunk 3 above)
Comment (Hipska) [reply Formula/telegraf.rb:27]: In Debian, you use the official builds: https://docs.influxdata.com/telegraf/v1.21/introduction/installation/

File: Formula/telegraf.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/telegraf.rb:7]: What about it? It doesn't say all those cases need a revision bump, it's just setting the scene.

Comment (Hipska) [reply Formula/telegraf.rb]: Okay thanks, I must have misread then.

File: Formula/telegraf.rb (hunk 2 above)
Comment (Hipska) [reply Formula/telegraf.rb]: But removed it..

**Review by bayandin — APPROVED** [review]
Thanks @Hipska 🎉

**Review by SMillerDev — APPROVED** [review]

**Comment by Hipska** [comment]
What a strange way to close PRs with unmerged commits and commit the merge separately.. But okay, thanks for the quick followup.

**Comment by Hipska** [comment]
For example, now this one doesn't count for my [hacktoberfest](https://github.com/topics/hacktoberfest) goals. If PRs get closed, what is the point on having the topic on this repository @SMillerDev @bayandin ?

**Comment by SMillerDev** [comment]
That's because hacktoberfest changed their counting method. It needs a label on the PR now too. TestBot squashing your commits had nothing to do with it.

**Comment by Hipska** [comment]
Thanks for adding the label!

It is indeed needed to add the label when the PR isn't accepted or merged. It still feels strange as I also don't appear on the [contributors](https://github.com/Homebrew/homebrew-core/graphs/contributors) page. It seems like credits aren't given the correct way because of this workflow, and yes I have seen the mention on d4aad3f.

**Pull request closed without merging**